curl: add -h cheat-sheet using curl-cheat-sheet as text (#18071)
curl/curl: docs/cmdline-opts/help.md

If no argument is provided, curl displays the most important command line
arguments.

The argument can either be a **category**, a **command line option**, or
**cheat-sheet**. When a category is provided, curl shows all command line
options within the given category. Specify category `all` to list all
available options.

If `category` is specified, curl displays all available help categories.

> If `cheat-sheet` is specified, curl displays a helpful cheat sheet.

Reviewer: Is everyone going to know the phrase cheat sheet?

```suggestion
If `cheat-sheet` is specified, curl displays the most common options.
```
